crates/egui_plot/src/lib.rs

```diff
@@ -689,15 +689,18 @@ impl Plot {
     }
 
     /// Interact with and add items to the plot and finally draw it.
-    pub fn show<R>(self, ui: &mut Ui, build_fn: impl FnOnce(&mut PlotUi) -> R) -> PlotResponse<R> {
-        self.show_dyn(ui, Box::new(build_fn))
-    }
-
-    fn show_dyn<'a, R>(
-        self,
-        ui: &mut Ui,
-        build_fn: Box<dyn FnOnce(&mut PlotUi) -> R + 'a>,
-    ) -> PlotResponse<R> {
+    pub fn show<'data, FN, R>(self, ui: &mut Ui, build_fn: FN) -> PlotResponse<R>
+    where
+        FN: FnOnce(&mut PlotUi<'data>) -> R,
+    {
+        //     self.show_dyn(ui, Box::new(build_fn))
+        // }
+
+        // fn show_dyn<'a, R>(
+        //     self,
+        //     ui: &mut Ui,
+        //     build_fn: Box<dyn FnOnce(&mut PlotUi) -> R + 'a>,
+        // ) -> PlotResponse<R> {
```

Comment:
I wonder about this one. why do we need to box the one-time callback?

Reply:
It's to reduce the amount of generated code by making sure that the plotting function can be reused.

Reply:
understood, will undo the change